qiita_plugins/target_gene/tgp/tests/test_qiita_client.py

```diff
@@ -0,0 +1,108 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+from os.path import exists
+from os import remove, environ, close
+from tempfile import mkstemp
+
+import httpretty
+
+from tgp.qiita_client import QiitaClient
+
+
+class QiitaClientTests(TestCase):
+    def setUp(self):
+        self.tester = QiitaClient("https://test_server.com")
+        self._clean_up_files = []
+
+    def tearDown(self):
+        for fp in self._clean_up_files:
+            if exists(fp):
+                remove(fp)
+
+    def test_init(self):
+        obs = QiitaClient("https://test_server.com")
+        self.assertEqual(obs._server_url, "https://test_server.com")
+        self.assertTrue(obs._verify)
+
+    def test_init_cert(self):
+        fd, conf_fp = mkstemp()
+        close(fd)
+        with open(conf_fp, 'w') as f:
+            f.write(CONF_FP)
```

Comment:
No, I don't want to change the values of the config file specifically, I want to make sure that the `__init__` functionality is parsing the correct config file. Note that in Qiita we have a configuration manager that is parsing such config and providing the information to the rest of the system. Here, the configuration file only affects this class, so the class can be though as the configuration manager itself.

If I use the same strategy that you are proposing, I still don't know if the class is parsing the config file in the environmental variable or the default config file in the repository.